src/test/kotlin/uk/gov/justice/digital/hmpps/personrecord/config/MessagingMultiNodeTestBase.kt

```diff
@@ -352,44 +352,6 @@ abstract class MessagingMultiNodeTestBase : IntegrationTestBase() {
       PurgeQueueRequest.builder().queueUrl(reclusterEventsQueue!!.dlqUrl).build(),
     )
 
-    await.atMost(Duration.ofSeconds(2)) untilCallTo {
-      probationEventsQueue!!.sqsClient.countAllMessagesOnQueue(probationEventsQueue!!.queueUrl).get()
-    } matches { it == 0 }
-    await.atMost(Duration.ofSeconds(2)) untilCallTo {
-      probationEventsQueue!!.sqsDlqClient!!.countAllMessagesOnQueue(probationEventsQueue!!.dlqUrl!!).get()
-    } matches { it == 0 }
-
-    await.atMost(Duration.ofSeconds(2)) untilCallTo {
-      probationMergeEventsQueue!!.sqsClient.countAllMessagesOnQueue(probationMergeEventsQueue!!.queueUrl).get()
-    } matches { it == 0 }
-    await.atMost(Duration.ofSeconds(2)) untilCallTo {
-      probationMergeEventsQueue!!.sqsDlqClient!!.countAllMessagesOnQueue(probationMergeEventsQueue!!.dlqUrl!!).get()
-    } matches { it == 0 }
-
-    await.atMost(Duration.ofSeconds(2)) untilCallTo {
-      probationDeleteEventsQueue!!.sqsClient.countAllMessagesOnQueue(probationDeleteEventsQueue!!.queueUrl).get()
-    } matches { it == 0 }
-    await.atMost(Duration.ofSeconds(2)) untilCallTo {
-      probationDeleteEventsQueue!!.sqsDlqClient!!.countAllMessagesOnQueue(probationDeleteEventsQueue!!.dlqUrl!!).get()
-    } matches { it == 0 }
-
-    await.atMost(Duration.ofSeconds(2)) untilCallTo {
-      prisonEventsQueue!!.sqsClient.countAllMessagesOnQueue(prisonEventsQueue!!.queueUrl).get()
-    } matches { it == 0 }
-    await.atMost(Duration.ofSeconds(2)) untilCallTo {
-      prisonEventsQueue!!.sqsDlqClient!!.countAllMessagesOnQueue(prisonEventsQueue!!.dlqUrl!!).get()
-    } matches { it == 0 }
-
-    await.atMost(Duration.ofSeconds(2)) untilCallTo {
-      prisonMergeEventsQueue!!.sqsClient.countAllMessagesOnQueue(prisonMergeEventsQueue!!.queueUrl).get()
-    } matches { it == 0 }
-    await.atMost(Duration.ofSeconds(2)) untilCallTo {
-      prisonMergeEventsQueue!!.sqsDlqClient!!.countAllMessagesOnQueue(prisonMergeEventsQueue!!.dlqUrl!!).get()
-    } matches { it == 0 }
-
-    await.atMost(Duration.ofSeconds(2)) untilCallTo {
-      reclusterEventsQueue!!.sqsClient.countAllMessagesOnQueue(reclusterEventsQueue!!.queueUrl).get()
-    } matches { it == 0 }
```

Comment:
It's probably OK if we just check the last one  we purged is empty - all the others should be empty by this point

Reply:
I know its inheritance again in another level, but could we do something like e.g. `UnmergeMessageingMultiNodeBase` where we only purge the queues we need to rather than all of them 🤔

Reply:
🤔 I don't mind inheritance in tests if it speeds things up.